x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -122,20 +124,64 @@ const MemoizedChart = React.memo(XYChart);
 
 export function XYChartReportable(props: XYChartRenderProps) {
   const [isReady, setIsReady] = useState(false);
```

Comment:
I generally like having only one useState per component, which would be pretty simple to do in this case. Not a big deal, though.